server/src/main.rs

```diff
@@ -0,0 +1,45 @@
+use actix_files as fs;
+use actix_web::{middleware, web, App, Error, HttpRequest, HttpResponse, HttpServer};
+use actix_web_actors::ws;
+use actix::{Actor, Addr};
+
+mod game;
+mod engine;
+
+async fn ws_index(r: HttpRequest, stream: web::Payload, data: web::Data<Addr<engine::Server>>) -> Result<HttpResponse, Error> {
+    ws::start(engine::WebSocket::new(data.get_ref().clone()), &r, stream)
+}
+
+fn env_or_default(env_name: &str, default: &str) -> String {
```

Comment:
```rust
fn env_or_default(env_name: &str, default: &str) -> String {
    std::env::var(env_name).unwrap_or(default.to_string())
}
```
might even consider dropping it completely ;)

(btw, these are the cases where I think `match` would be better)

Reply:
Yeah, this section could be better. Created issue for the `Config` #2